numba/_lapack.c

```diff
@@ -1019,21 +1116,179 @@ numba_ez_gesdd(char kind, char jobz, Py_ssize_t m, Py_ssize_t n, void *a,
                Py_ssize_t lda, void *s, void *u, Py_ssize_t ldu, void *vt,
                Py_ssize_t ldvt)
 {
+    ENSURE_VALID_KIND(kind)
+
     switch (kind)
     {
         case 's':
         case 'd':
-            return numba_ez_rgesdd(kind, jobz, m, n, a, lda, s, u, ldu, vt, ldvt);
+            return numba_ez_rgesdd(kind, jobz, m, n, a, lda, s, u, ldu, vt,
+                                   ldvt);
         case 'c':
         case 'z':
-            return numba_ez_cgesdd(kind, jobz, m, n, a, lda, s, u, ldu, vt, ldvt);
-        default:
-        {
-            PyGILState_STATE st = PyGILState_Ensure();
-            PyErr_SetString(PyExc_ValueError,\
-                            "invalid kind in numba_ez_gesdd call");
-            PyGILState_Release(st);
-        }
+            return numba_ez_cgesdd(kind, jobz, m, n, a, lda, s, u, ldu, vt,
+                                   ldvt);
+    }
+    return -1; // unreachable
+}
+
+
+/*
+ * Compute the QR factorization of a matrix.
+ * Return -1 on internal error, 0 on success, > 0 on failure.
+ */
+static int
+numba_raw_xgeqrf(char kind, Py_ssize_t m, Py_ssize_t n, void *a, Py_ssize_t
+                 lda, void *tau, void *work, Py_ssize_t lwork, Py_ssize_t *info)
+{
+    void *raw_func = NULL;
+    F_INT _m, _n, _lda, _lwork, _info;
+
+    ENSURE_VALID_KIND(kind)
+
+    switch (kind)
+    {
+        case 's':
+            raw_func = get_clapack_sgeqrf();
+            break;
+        case 'd':
+            raw_func = get_clapack_dgeqrf();
+            break;
+        case 'c':
+            raw_func = get_clapack_cgeqrf();
+            break;
+        case 'z':
+            raw_func = get_clapack_zgeqrf();
+            break;
+    }
+    if (raw_func == NULL)
```

Comment:
@pitrou We also have a lot of these, if the getter fails for some reason, do you know where the exception is set for raising?